cypress-io/cypress
# fix: Remove unneeded Firefox headless hack (run ci)

- Closes https://github.com/cypress-io/cypress/issues/24900

### User facing changelog
Fixes an issue where recent versions of Firefox in headless mode would behave inconsistently with headed.

### Additional details
Back in 2021, Firefox had a quirk where in headless mode it would never gain focus; a workaround was added in https://github.com/cypress-io/cypress/commit/8a4848837aea88f79446c77eeac097b713a0d0a6, forcing the active tab into focus whenever an element in the document gained focus.

However, this results in inconsistent behavior between headed and headless mode in some circumstances. For example, a user might click on a button to open a dropdown, but then Cypress forces focus onto the tab as a whole, causing the dropdown to close again before it can be interacted with.

The workaround no longer appears to be necessary in modern versions of Firefox; both locally and in CI, removing it results in consistent and working behavior.

### Steps to test
See https://github.com/cypress-io/cypress/issues/24900 for a repro. Contact @BlueWinds if you want to run this example locally, she has the login information from the reporter.

### How has the user experience changed?
No change; behavior is now consistent between headless and headed mode.

### PR Tasks
- [n/a] Have tests been added/updated?
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
